src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -170,29 +169,37 @@ void SQLiteDBEngine::syncTableRowData(const std::string& table,
     {
         if (getPrimaryKeysFromTable(table, primaryKeyList))
         {
-            ReturnTypeCallback resultCbType{ MODIFIED };
-            nlohmann::json jsResult;
-            const bool diffExist { getRowDiff(primaryKeyList, table, data.at(0), jsResult) };
-            if (diffExist)
+            nlohmann::json bulkInsertJson;
+            for (const auto& entry : data)
             {
-                const nlohmann::json jsDataToUpdate{getDataToUpdate(primaryKeyList, jsResult, data.at(0), inTransaction)};
-                if (!jsDataToUpdate[0].empty())
+                nlohmann::json jsResult;
+                const bool diffExist { getRowDiff(primaryKeyList, table, entry, jsResult) };
+                if (diffExist)
                 {
-                    const auto& transaction { m_sqliteFactory->createTransaction(m_sqliteConnection)};
-                    updateSingleRow(table, jsDataToUpdate[0]);
-                    transaction->commit();
+                    const nlohmann::json jsDataToUpdate{getDataToUpdate(primaryKeyList, jsResult, entry, inTransaction)};
+                    if (!jsDataToUpdate[0].empty())
+                    {
+                        const auto& transaction { m_sqliteFactory->createTransaction(m_sqliteConnection)};
```

Comment:
this transaction is necessary ?

Reply:
for the statement bind within updateSingleRow